doc/rados/troubleshooting/log-and-debug.rst

```diff
@@ -2,177 +2,204 @@
+
+   sudo ceph daemon osd.0 config set debug_osd 0/5
+
+For details on available settings, see `Subsystem, Log and Debug Settings`_.
 
 
 Boot Time
 =========
 
-To activate Ceph's debugging output (*i.e.*, ``dout()``) at boot time, you must
-add settings to your Ceph configuration file. Subsystems common to each daemon
-may be set under ``[global]`` in your configuration file. Subsystems for
-particular daemons are set under the daemon section in your configuration file
-(*e.g.*, ``[mon]``, ``[osd]``, ``[mds]``). For example
+To activate Ceph's debugging output (that is, the ``dout()`` logging function)
+at boot time, you must add settings to your Ceph configuration file.
+Subsystems that are common to all daemons are set under ``[global]`` in the
+configuration file. Subsystems for a specific daemon are set under the relevant
+daemon section in the configuration file (for example, ``[mon]``, ``[osd]``,
+``[mds]``). Here is an example that shows possible debugging settings in a Ceph
+configuration file:
 
 .. code-block:: ini
 
-	[global]
-		debug_ms = 1/5
-		
-	[mon]
-		debug_mon = 20
-		debug_paxos = 1/5
-		debug_auth = 2
-		 
- 	[osd]
- 		debug_osd = 1/5
- 		debug_filestore = 1/5
- 		debug_journal = 1
- 		debug_monc = 5/20
- 		
-	[mds]
-		debug_mds = 1
-		debug_mds_balancer = 1
+    [global]
+        debug_ms = 1/5
+        
+    [mon]
+        debug_mon = 20
+        debug_paxos = 1/5
+        debug_auth = 2
+         
+     [osd]
+         debug_osd = 1/5
+         debug_filestore = 1/5
+         debug_journal = 1
+         debug_monc = 5/20
+         
+    [mds]
+        debug_mds = 1
+        debug_mds_balancer = 1
 
 
-See `Subsystem, Log and Debug Settings`_ for details.
+For details, see `Subsystem, Log and Debug Settings`_.
 
 
 Accelerating Log Rotation
 =========================
 
-If your OS disk is relatively full, you can accelerate log rotation by modifying
-the Ceph log rotation file at ``/etc/logrotate.d/ceph``. Add  a size setting
-after the rotation frequency to accelerate log rotation (via cronjob) if your
-logs exceed the size setting. For example, the  default setting looks like
-this::
-   
-	rotate 7
-  	weekly
-  	compress
-  	sharedscripts
-   	
-Modify it by adding a ``size`` setting. ::
-   
-  	rotate 7
-  	weekly
-  	size 500M
-  	compress
-  	sharedscripts
-
-Then, start the crontab editor for your user space. ::
-   
-  	crontab -e
-	
-Finally, add an entry to check the ``etc/logrotate.d/ceph`` file. ::
-   
-  	30 * * * * /usr/sbin/logrotate /etc/logrotate.d/ceph >/dev/null 2>&1
-
-The preceding example checks the ``etc/logrotate.d/ceph`` file every 30 minutes.
+If your OS disk is nearly full, you can accelerate log rotation by modifying
+the Ceph log rotation file at ``/etc/logrotate.d/ceph``. To use cronjob to
+accelerate log rotation whenever logs exceed a specified size, add a size
+setting after the rotation-frequency setting. The procedure for adding the size
+setting can be illustrated as follows. 
+
```

Comment:
To increase log rotation frequency and thus guard against filesystem fillage, add a ``size`` directive after the `weekly` frequency directive.  You might also consider changing ``weekly`` to ``daily`` and ``rotate`` to ``30`` to additionally smooth out volume spikes.

Reply:
Accepted (with some alterations so that this suggestion fits into the Procrustean bed of the extant text).

Reply:
Second time in 24 hours I've seen _procrustean_.  hmmm.